### PR TITLE
[apmpackage] fix booleans for package upgrade

### PR DESCRIPTION
**Pull request by stuartnelson3**
fixes https://github.com/elastic/kibana/issues/122192

According to @ogupte:

>the apmpackage defines a handlebars template to render the final apm server config, but it omits whole sections of configs when certain bools are false : these include `enable_rum`, `tls_enabled`, and `tail_sampling_policies`. the package should be updated to rendered these complete config sections regardless of these boolean values.

### CODE REVIEW

**Comment by stuartnelson3** [comment]
Question: Is it too late to backport to `7.16` and make it for the `7.16.3` release?

File: apmpackage/apm/agent/input/template.yml.hbs [hunk 1]
```diff
@@ -65,14 +62,11 @@ apm-server:
         {{#each tls_curve_types}}
           - {{this}}
         {{/each}}
-    {{/if}}
     write_timeout: {{write_timeout}}
-    {{#if tail_sampling_policies}}
     sampling:
       tail:
         enabled: true
```
Comment (marclop) [inline apmpackage/apm/agent/input/template.yml.hbs:68]: won't this cause tail based sampling to always be enabled? we should probably infer the value on whether the `tail_sampling_policies` has any policies.

Comment (simitt) [reply apmpackage/apm/agent/input/template.yml.hbs:68]: +

Comment (stuartnelson3) [reply apmpackage/apm/agent/input/template.yml.hbs]: updated, thanks!

**Review by marclop — APPROVED** [review]

**Comment by axw** [comment]
Tested with apm-integration-testing, 8.1.0 BC2.

With RUM, TLS, and TBS all disabled, the policy is rendered with the relevant settings blocks:

```yaml
    apm-server:
      capture_personal_data: true
      max_connections: 0
      max_event_size: 307200
      auth:
        api_key:
          enabled: false
          limit: 100
        anonymous:
          enabled: false
          allow_agent:
            - rum-js
            - js-base
            - iOS/swift
          allow_service: null
          rate_limit:
            ip_limit: 1000
            event_limit: 300
        secret_token: null
      default_service_environment: null
      shutdown_timeout: 30s
      sampling:
        tail:
          enabled: false
          policies:
            - sample_rate: 0.11
            - trace.name: foo
              sample_rate: 0.5
          interval: 2m
      rum:
        enabled: false
        exclude_from_grouping: ^/webpack
        allow_headers: null
        response_headers: null
        library_pattern: node_modules|bower_components|~
        allow_origins:
          - '*'
        source_mapping:
          metadata: []
      ssl:
        enabled: false
        key_passphrase: null
        certificate: null
        supported_protocols:
          - TLSv1.0
          - TLSv1.1
          - TLSv1.2
        curve_types: null
        key: null
        cipher_suites: null
      response_headers: null
      write_timeout: 30s
      host: 'localhost:8200'
      max_header_size: 1048576
      idle_timeout: 45s
      expvar.enabled: false
      read_timeout: 3600s
      java_attacher:
        enabled: false
        discovery-rules: null
        download-agent-version: null
      agent_config: []
```

**Pull request merged**